common/web/input-processor/src/text/prediction/predictionContext.ts

```diff
@@ -37,93 +31,84 @@ export default class PredictionContext extends EventEmitter<PredictionContextEve
   private doRevert: boolean = false;
   private recentRevert: boolean = false;
 
-
   private langProcessor: LanguageProcessor;
+  private kbdProcessor: KeyboardProcessor;
+
+  /**
+   * Represents the active context used when requesting and applying predictive-text operations.
+   */
+  private _currentTarget: OutputTarget;
+
+  public get currentTarget(): OutputTarget {
+    return this._currentTarget;
+  }
+
+  public setCurrentTarget(target: OutputTarget): Promise<Suggestion[]> {
+    const originalTarget = this._currentTarget;
+    this._currentTarget = target;
+
+    if(originalTarget != target) {
+      // Note:  should be triggered after the corresponding new-context event rule has been processed,
+      // as that may affect the value of layerId here.
+      return this.langProcessor.invalidateContext(target, this.kbdProcessor.layerId);
+    } else {
+      return Promise.resolve([]);
+    }
+  }
 
   private readonly suggestionApplier: (suggestion: Suggestion) => Promise<Reversion>;
   private readonly suggestionReverter: (reversion: Reversion) => void;
 
   /**
    * Handler for post-processing once a suggestion has been applied: calls
    * into the active keyboard's `begin postKeystroke` entry point.
+   *
+   * Called after the suggestion is applied but _before_ new predictions are
+   * requested based on the resulting context.
    */
-  private readonly postApplicationHandler: (target: OutputTarget) => void;
+  private readonly postApplicationHandler: () => void;
 
-  public constructor(langProcessor: LanguageProcessor, kbdProcessor: KeyboardProcessor, target: OutputTarget) {
+  public constructor(langProcessor: LanguageProcessor, kbdProcessor: KeyboardProcessor) {
     super();
 
     if(langProcessor.state == 'inactive') {
       throw new Error("Invalid state:  no predictive-text model is currently available.");
     }
+    this.langProcessor = langProcessor;
+    this.kbdProcessor = kbdProcessor;
 
-    if(!target) {
-      throw new Error("No text context is available to use for prediction!");
+    const validSuggestionState: () => boolean = () => {
+      return this.currentTarget && langProcessor.state == 'configured';
     }
```

Comment:
```suggestion
    const validSuggestionState: () => boolean = () => 
      this.currentTarget && langProcessor.state == 'configured';
```

Should also work yeah? No big deal though